### PR TITLE
[wil] Update to v1.0.220914.1

### PR DESCRIPTION
**Pull request by CookiePLMonster**
- #### What does your PR fix?
  Updates WIL to https://github.com/microsoft/wil/releases/tag/v1.0.220914.1. Also adds natvis installation as WIL added one recently, and updates the copyright installation to use a "modern" function.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  triplets unchanged, baseline not modified

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

File: ports/wil/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "wil",
-  "version-date": "2021-12-25",
+  "version-date": "2022-09-23",
   "description": "The Windows Implementation Libraries (WIL) is a header-only C++ library created to make life easier for developers on Windows through readable type-safe C++ interfaces for common Windows coding patterns.",
   "homepage": "https://github.com/microsoft/wil",
```
Comment (MonicaLiu0311) [inline ports/wil/vcpkg.json:5]:
```suggestion
  "homepage": "https://github.com/microsoft/wil",
  "license": "MIT",
```

Comment (MonicaLiu0311) [reply ports/wil/vcpkg.json:5]: A license is required.

Comment (CookiePLMonster) [reply ports/wil/vcpkg.json:5]: Added.

**Comment by CookiePLMonster** [comment]
Updated to add the license definition.

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**